### PR TITLE
FISH-128 OpenAPI does not include APIs from jars within a war 

### PR DESCRIPTION
**Pull request by jGauravGupta**
Signed-off-by: Gaurav Gupta <gaurav.gupta@payara.fish>

## Description
This is a feature to include classes for OpenAPI Metadata scanning from the packaged jar within a  war.

### Testing Performed
Unit tests, Manual testing, MP OpenAPI TCK Runner

### Testing Environment
Windows 10, JDK 11.0

## Documentation
TODO

### CODE REVIEW

**Comment by jGauravGupta** [comment]
jenkins test please

**Comment by MattGill98** [comment]
Needs merge conflicts resolving

**Comment by jGauravGupta** [comment]
jenkins test please

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java [hunk 1]
```diff
@@ -262,6 +271,52 @@ private OpenAPI buildDocument() throws OpenAPIBuildException {
...
+
+        Set<Type> types = new HashSet<>();
+        if (config != null && config.getScanLib()) {
+            Iterator<String> subArchiveItr = archive.entries().asIterator();
+            while (subArchiveItr.hasNext()) {
+                String subArchive = subArchiveItr.next();
+                if (subArchive.startsWith("WEB-INF/lib/") && subArchive.endsWith(".jar")) {
+                    try {
+                        Iterator<String> classItr = archive.getSubArchive(subArchive).entries().asIterator();
+                        while (classItr.hasNext()) {
```
Comment (jbee) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java:288]: I think you could just use a `for` loop with iterator here.

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java [hunk 2]
```diff
@@ -262,6 +271,52 @@ private OpenAPI buildDocument() throws OpenAPIBuildException {
...
+    /**
+     * @return a list of all classes in the archive.
+     */
+    private Set<Type> filterTypes(ApplicationInfo appInfo, OpenApiConfiguration config) {
+        ReadableArchive archive = appInfo.getSource();
+
+        Set<Type> types = new HashSet<>();
+        if (config != null && config.getScanLib()) {
+            Iterator<String> subArchiveItr = archive.entries().asIterator();
+            while (subArchiveItr.hasNext()) {
```
Comment (jbee) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java:283]: I think you could just use a `for` loop with iterator here.

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java [hunk 3]
```diff
@@ -262,6 +271,52 @@ private OpenAPI buildDocument() throws OpenAPIBuildException {
 
     }
 
+    /**
+     * @return a list of all classes in the archive.
+     */
+    private Set<Type> filterTypes(ApplicationInfo appInfo, OpenApiConfiguration config) {
```
Comment (jbee) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java:277]: Quite a big method that could be split into a few steps.

**Comment by jGauravGupta** [comment]
jenkins test please

**Comment by jGauravGupta** [comment]
Test War file with packaged Jar including OpenAPI `@Schema` classes

**Comment by jGauravGupta** [comment]
jenkins test please

**Pull request merged**